lib/scales.mjs

```diff
@@ -0,0 +1,234 @@
+/*
+ * Adapted from https://github.com/georgedoescode/fluid-design-system-on-demand-builders
+ * …which in turn is based on https://utopia.fyi
+ */
+
+export const ratios = {
+  'minor-second': 1.067,
+  'major-second': 1.125,
+  'minor-third': 1.2,
+  'major-third': 1.25,
+  'perfect-fourth': 1.333,
+  'augmented-fourth': 1.414,
+  'perfect-fifth': 1.5,
+  'golden-ratio': 1.618,
+  'major-sixth': 1.667,
+  'minor-seventh': 1.778,
+  'major-seventh': 1.875,
+  octave: 2,
+}
+
+const defaultConfig = {
+  steps: 6,
+  viewportMin: 320,
+  viewportMax: 1500,
+  baseMin: 16,
+  baseMax: 18,
+  scaleMin: 'minor-third',
+  scaleMax: 'perfect-fourth',
+}
+
+/**
+ * @param {(string|number)} ratio - A named ratio or an explicit ratio
+ * @param {string} scaleProperty - The scale property the ratio value will be used for; used in error reporting only
+ * @returns {number} The numeric ratio value
+ */
+export function getRatioValue(ratio, scaleProperty) {
+  // Config provided a raw number for the ratio
+  if (typeof ratio === 'number') return ratio
+
+  // Config provided a named value for the ratio
+  if (Object.hasOwn(ratios, ratio)) return ratios[ratio]
+
+  // Sad trombone
+  throw new Error(`Value provided for ${scaleProperty} must be a number or a named ratio: ${Object.keys(ratios).join(', ')}`)
+}
+
+// Rounds to the nearest 2 decimal places
+// e.g. roundToHundredths(4.894) = 4.89
+//      roundToHundredths(4.896) = 4.90
+/**
+ * @param {number} num - The number to apply rounding to
+ * @returns {number} The supplied number rounded to the nearest hundredths decimal place
+ */
+export function roundToHundreths(num) {
+  return Math.round(num * 100) / 100
+}
+
+/**
+ * @param {Object} config - The configuration object
+ * @param {number} config.baseMinPx - The base size, in pixels, at the minimum viewport width
+ * @param {number} config.baseMaxPx - The base size, in pixels, at the maximum viewport width
+ * @param {number} config.viewportMinPx - The width, in pixels, of the minimum viewport
+ * @param {number} config.viewportMaxPx - The width, in pixels, of the maximum viewport
+ * @returns {string} The CSS `clamp` value to be used
+ */
+export function generateClamp({
+  baseMinPx,
+  baseMaxPx,
+  viewportMinPx,
+  viewportMaxPx,
+}) {
+  const pixelsPerRem = 16
```

Comment:
That's true, although the new way to set the base font size is with `config.typeScale.baseMin` and `config.typeScale.baseMax`. So they could set `baseMin: 14` and `baseMax: 16` for example if they want a smaller base. Maybe this is something to address more clearly in the readme & docs?

(Also, I'm wondering if the amount of folks setting their own base font size on the body is because previous to this, Enhance Styles defaulted to 18?)

Reply:
Yes. Since the tendency is to manually set a base font size on the body then I agree we should call this out in the docs.

Reply:
Sounds good, will do!